### DISCUSSION TITLE
GLTF 2.0 Animaiton channel conversion

### DISCUSSION THREAD

**Discussion by hypnosnhendricks** [assimp/assimp#3547] [General]
In the GLTF 2.0 spec, Animation channels for rotation & weights can be non-float. 
See https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#animations

AssImp currently has no support for this. Is there any work in progress being done to support the non-float conversion? If not, are there any plans to support it? 

I was curious as if it wasn't in progress or planned, I might take it on as I would want to also use this to support mesh quantizaiton in assimp as spec'd here. 
https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_mesh_quantization

**Comment by kimkulling** [comment]
Is there any advantage in having these data in non-float?

**Reply by hypnosnhendricks** [reply]
One of our clients uses assimp and I just happen to notice that assimp doesn't seem to meet the GLTF 2.0 spec here. The ask was to support animation & mesh quantization in our GLB delivery to clients, so I'm looking at what toolchains our various clients use to see which ones support what extensions & if they don't, how much work would it be to implement it.

Looking at the assimp code, it doesn't seem like it would be to much work to support animation quantization. The mesh quantization is trickier as it would require the extra node and skin matrix information & it wasn't obvious how the assimp standard model would contain that information.

**Comment by kimkulling** [comment]
Looks like we could do this without rewrite the whole stuff. Is there already a feature-request in the issue-db?

**Reply by hypnosnhendricks** [reply]
I created one specifically for animaiton. https://github.com/assimp/assimp/issues/3560.
I didn't create one for KHR_mesh_quantization as that is a separate topic.

**Reply by hypnosnhendricks** [reply]
Also, a write up on various mesh compression issues here: https://github.com/KhronosGroup/glTF/issues/588#issuecomment-750402206

**Reply by kimkulling** [reply]
Thanks for the link! I will read it.
